File: app/src/main/java/com/xadaptersimple/MainActivity.java
package com.xadaptersimple;

import android.content.Intent;
import android.os.Bundle;
import android.support.v7.app.AppCompatActivity;
import android.view.View;

public class MainActivity extends AppCompatActivity implements View.OnClickListener {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        findViewById(R.id.network).setOnClickListener(this);
        findViewById(R.id.LinearLayout).setOnClickListener(this);
        findViewById(R.id.GridLayout).setOnClickListener(this);
        findViewById(R.id.StaggeredGridLayout).setOnClickListener(this);
        findViewById(R.id.collapsingToolbar).setOnClickListener(this);
        findViewById(R.id.emptyView).setOnClickListener(this);
        findViewById(R.id.SwipeRefreshLayout).setOnClickListener(this);
    }

    @Override
    public void onClick(View view) {
        switch (view.getId()) {
            case R.id.network:
                startActivity(NetWorkActivity.class);
                break;
            case R.id.LinearLayout:
                startActivity(LinearLayoutManagerActivity.class);
                break;
            case R.id.GridLayout:
                startActivity(GridLayoutManagerActivity.class);
                break;
            case R.id.StaggeredGridLayout:
                startActivity(StaggeredGridLayoutManagerActivity.class);
                break;
            case R.id.collapsingToolbar:
                startActivity(CollapsingToolbarLayoutActivity.class);
                break;
            case R.id.emptyView:
                startActivity(EmptyViewActivity.class);
                break;
            case R.id.SwipeRefreshLayout:
                startActivity(SwipeRefreshLayoutActivity.class);
                break;
        }
    }

    private void startActivity(Class<?> clz) {
        Intent intent = new Intent(getApplicationContext(), clz);
        intent.setFlags(Intent.FLAG_ACTIVITY_NEW_TASK);
        startActivity(intent);
    }
}

File: app/src/main/java/com/xadaptersimple/MainActivity.kt
package com.xadaptersimple

import android.content.Intent
import android.os.Bundle
import androidx.appcompat.app.AppCompatActivity
import com.xadaptersimple.net.NetApi
import com.xadaptersimple.net.NetWorkActivity
import io.reactivex.network.RxNetWork
import io.reactivex.network.SimpleRxNetOptionFactory
import kotlinx.android.synthetic.main.activity_main.*
import retrofit2.converter.gson.GsonConverterFactory

class MainActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)

        RxNetWork.initialization(SimpleRxNetOptionFactory(NetApi.ZL_BASE_API, GsonConverterFactory.create()))

        setContentView(R.layout.activity_main)
        network.setOnClickListener { startActivity(NetWorkActivity::class.java) }
        LinearLayout.setOnClickListener { startActivity(LinearLayoutManagerActivity::class.java) }
        GridLayout.setOnClickListener { startActivity(GridLayoutManagerActivity::class.java) }
        StaggeredGridLayout.setOnClickListener { startActivity(StaggeredGridLayoutManagerActivity::class.java) }
        collapsingToolbar.setOnClickListener { startActivity(CollapsingToolbarLayoutActivity::class.java) }
        emptyView.setOnClickListener { startActivity(EmptyViewActivity::class.java) }
        multiple.setOnClickListener { startActivity(MultipleXXItemActivity::class.java) }
        test.setOnClickListener { startActivity(TestActivity::class.java) }
        refreshLayout.setOnClickListener { startActivity(RefreshLayoutActivity::class.java) }
        databinding.setOnClickListener { startActivity(DataBindingActivity::class.java) }
    }

    private fun startActivity(clz: Class<*>) {
        val intent = Intent(applicationContext, clz)
        intent.flags = Intent.FLAG_ACTIVITY_NEW_TASK
        startActivity(intent)
    }
}
